src/PolynomialSolutions.jl

```diff
@@ -108,6 +109,34 @@ function multiply_by_r(p::Polynomial{N,T}, k::Int) where {N,T}
     return multiply_by_r(q, k - 2)
 end
 
+"""
+    multiply_by_anisotropic_r(A::SMatrix{N,N,T}, p::Polynomial, k::Int = 2)
+
+Multiply a polynomial `p` by the monomial `r_A^k`, where `r_A = |r^T A^{-1} r|`,
+r = (x_1, x_2, ... x_n]) and `k` is an even positive integer.
+"""
+# TODO While StaticArrays provides Rational support for `inv`, the problem is that Float64 gets promoted to BigFloat
+# when computing A^{-1} with Float64. Is there a way to get the best of both worlds?
```

Comment:
It is not clear what exactly you want here. Also, why does `A` have to be of `Float64`?

Reply:
This was a little silly, because I didn't understand what I was doing properly. I think all I realize how to fix all the issues I didn't know how to before. I'm still not sure, but I am closer.

The current problem (and why tests fail in the next commit) is that types like `Rational{Int64}` will get promoted to `Rational{BigInt}`, which either results in imprecision or in a type error.